packages/block-editor/src/components/block-inspector/index.js

```diff
@@ -32,19 +32,37 @@ import { useBorderPanelLabel } from '../../hooks/border';
 
 import { unlock } from '../../lock-unlock';
 
+function BlockStylesPanel( { clientId } ) {
+	return (
+		<div>
+			<PanelBody title={ __( 'Styles' ) }>
+				<BlockStyles clientId={ clientId } />
+			</PanelBody>
+		</div>
+	);
+}
+
 function BlockInspectorLockedBlocks( { topLevelLockedBlock } ) {
-	const contentClientIds = useSelect(
+	const { contentClientIds, hasBlockStyles } = useSelect(
 		( select ) => {
 			const {
 				getClientIdsOfDescendants,
 				getBlockName,
 				getBlockEditingMode,
 			} = select( blockEditorStore );
-			return getClientIdsOfDescendants( topLevelLockedBlock ).filter(
-				( clientId ) =>
-					getBlockName( clientId ) !== 'core/list-item' &&
-					getBlockEditingMode( clientId ) === 'contentOnly'
-			);
+			const { getBlockStyles } = select( blocksStore );
+			return {
+				contentClientIds: getClientIdsOfDescendants(
+					topLevelLockedBlock
+				).filter(
+					( clientId ) =>
+						getBlockName( clientId ) !== 'core/list-item' &&
+						getBlockEditingMode( clientId ) === 'contentOnly'
+				),
```

Comment:
This will return a new array reference every time the block editor stores updates. It should also log a warning.

The previous implementation was a hack; returning just an array of string client IDs will pass a shallow equality check. Similar hacks should have inline comments to avoid regressions.

## New warning.
![CleanShot 2024-08-28 at 19 44 08](https://github.com/user-attachments/assets/1e2d8118-c66f-470b-bcc3-5023dc4b533f)

Reply:
Good catch! Thank you!